pkg/bot/slack.go

```diff
@@ -286,16 +289,22 @@ func (b *Slack) send(event slackMessage, req string, resp string) error {
 	}
 
 	options := []slack.MsgOption{
-		slack.MsgOptionText(formatx.CodeBlock(resp), false),
+		b.renderer.RenderInteractiveMessage(resp),
 	}
 
 	//if the message is from thread then add an option to return the response to the thread
 	if event.ThreadTimeStamp != "" {
 		options = append(options, slack.MsgOptionTS(event.ThreadTimeStamp))
 	}
 
-	if _, _, err := b.client.PostMessage(event.Channel, options...); err != nil {
-		return fmt.Errorf("while posting Slack message: %w", err)
+	if resp.OnlyVisibleForYou != "" {
+		if _, err := b.client.PostEphemeral(event.Channel, event.User, options...); err != nil {
+			return fmt.Errorf("while posting Slack message: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("while posting Slack message visible only to user: %w", err)
```

We use this “error wrapping” to be able to distinguish the error context, and currently you have the exact same message in 302 and 306 which will generate a problem during debugging.